great_expectations/core/expectation_validation_result.py

```diff
@@ -532,6 +532,12 @@ def __init__(  # noqa: PLR0913
         self.id = id
         self._metrics: dict = {}
 
+    @property
+    def asset_name(self) -> str | None:
+        if "active_batch_definition" in self.meta:
+            return self.meta["active_batch_definition"].data_asset_name
+        return None
```

Comment:
Would like to make this a top-level property (might write a ticket around it) but not pressing